### PR TITLE
Hotfix/remove annoying prints

### PR DESCRIPTION
**Pull request by morantr**
This PR fixes some print issues discovered on RDKB where using UDS_BUS, and print issue which appears also on Linux.

### CODE REVIEW

**Comment by vitalybu** [comment]
Do not merge this PR.
I want to examine the issue further.

File: controller/src/beerocks/master/tasks/task.cpp
Comment (tomereli) [inline controller/src/beerocks/master/tasks/task.cpp:49]: I don't get this. Each task is registering for events, right? And in the controller we pass events only to tasks which should receive these events. If this is not the case, it is a bug, and this commit is just hiding it instead of fixing it.
Please correct me if i'm wrong..

File: common/beerocks/btl/include/btl/btl.h
Comment (tomereli) [inline common/beerocks/btl/include/btl/btl.h]: Not acceptable. If you have a bug you solve it, not work around it to get the change merged.

File: common/beerocks/bcl/source/beerocks_socket_thread.cpp
Comment (tomereli) [inline common/beerocks/bcl/source/beerocks_socket_thread.cpp:187]: This completely breaks the abstraction! Why didn't you just add this to `btl_uds.cpp` before calling `handle_cmdu`? Why add it to the socket thread class? This whole commit is wrong.

File: common/beerocks/btl/btl_uds.cpp
Comment (tomereli) [inline common/beerocks/btl/btl_uds.cpp:128]: This function should be removed completely and replaced with a static function in the btl_uds.cpp implementation only.

File: common/beerocks/btl/CMakeLists.txt
Comment (tomereli) [inline common/beerocks/btl/CMakeLists.txt:22]: Good - so this should be a fixup commit.

**Review by tomereli — CHANGES_REQUESTED** [review]

File: controller/src/beerocks/master/tasks/task.cpp
Comment (morantr) [reply controller/src/beerocks/master/tasks/task.cpp:49]: > Each task is registering for events, right?

No.
A task doesn't register for anything. A task can decide that it is pending for an event to block its work cycle until the event is being received.
But, it is normal behavior that events would get to a task even if it is not pending for it.

File: common/beerocks/btl/include/btl/btl.h
Comment (morantr) [reply common/beerocks/btl/include/btl/btl.h]: Vitaly fixed it. I had difficulties to find out the root cause of the problem so I committed it so the change will be committed, knowing it will be fixed later.

File: common/beerocks/bcl/source/beerocks_socket_thread.cpp
Comment (morantr) [reply common/beerocks/bcl/source/beerocks_socket_thread.cpp:187]: Why do you say it is breaking the abstraction? This function is implemented on socket_thread, and can be overridden if needed.
The socket thread is base class for btl classes, and this function is needed for abstraction.
Anyway, it cannot be added to btl classes since I had to call it from a function implemented only on the socket thread, which I didn't want to override.

File: common/beerocks/btl/btl_uds.cpp
Comment (morantr) [reply common/beerocks/btl/btl_uds.cpp:128]: That is not possible since it is being called on socket_thread.

File: common/beerocks/btl/CMakeLists.txt
Comment (morantr) [reply common/beerocks/btl/CMakeLists.txt:22]: Sure, I will squash it before I'll mark it as ready for merge.

File: common/beerocks/bcl/include/bcl/beerocks_socket_thread.h
Comment (vitalybu) [inline common/beerocks/bcl/include/bcl/beerocks_socket_thread.h:45]: I don't like this name... Maybe change it to `should_handle_ieee1905_msg()`?

Comment (tomereli) [reply common/beerocks/bcl/include/bcl/beerocks_socket_thread.h:45]: OK I understand why it is needed, but please change the name to @vitalybu 's suggestion before merge

File: common/beerocks/btl/include/btl/btl.h
Comment (tomereli) [reply common/beerocks/btl/include/btl/btl.h]: You marked the PR "ready for merge". If you push a commit like that and plan to fix it, it is perfectly fine, but you need to mention that in the commit message that this commit should be dropped before merge, like @rmelotte does often.

File: controller/src/beerocks/master/tasks/task.cpp
Comment (tomereli) [reply controller/src/beerocks/master/tasks/task.cpp:49]: Well I think its not an ideal design choice, why is it implemented like that? Why can't a task "subscribe" to events? we can have the base class drop events for which the task didn't subscribe to. @vitalybu ?

File: common/beerocks/bcl/source/beerocks_socket_thread.cpp
Comment (tomereli) [reply common/beerocks/bcl/source/beerocks_socket_thread.cpp:187]: I got it all wrong, forget about it. Just change the name and don't mention UDS - perhaps on other bus types the filtering will also be implemented manually so just in the local bus case we skip this function.

**Review by tomereli — APPROVED** [review]

File: controller/src/beerocks/master/tasks/task.cpp
Comment (morantr) [reply controller/src/beerocks/master/tasks/task.cpp:49]: Why do you think that a task subscribing to an event is an ideal design? IMO it is not.
What is the benefit of such registration? Nothing.
It is not like when you push event, its reach to all the tasks and they have to filter it if they are subscribing to it or not. When we send an event we send it to a specific task ID, so only one instance of a specific task class would get it.
In the current design, each task has it is own event types, and on subscribing event design it would have to be global shared events for all the tasks, like TLVF, but that is just not necessary.

**Review by itayx — APPROVED** [review]

**Pull request merged**
